types/underscore/index.d.ts

```diff
@@ -781,12 +785,13 @@ declare module _ {
         drop: UnderscoreStatic['rest'];
 
         /**
-        * Returns a copy of the array with all falsy values removed. In JavaScript, false, null, 0, "",
-        * undefined and NaN are all falsy.
-        * @param array Array to compact.
-        * @return Copy of `array` without false values.
-        **/
-        compact<T>(array: _.List<T | null | undefined | false | "" | 0> | null | undefined): T[]
```

Comment:
No, `T` is in there at the beginning of `List`, I think this worked but took advantage of some interesting and possibly sketchy behavior where it seems that `T` was inferred as everything but those other types and was a clever solution for making this work better than it would have otherwise before conditional types were introduced. However, this solution doesn't work in the other interfaces, so I updated this to be consistent with their updates (and to move this away from the possible sketchiness).

Reply:
Oh, possibly you're referring to the `null | undefined` after the `List` declaration. That might have been a partial fix for #20302, which I considered including in this PR set but then passed on. I suppose it would be nice if I didn't break it again, though, but on the other hand it would probably be better for folks to check that they have something worth calling `compact` on before calling it.

Reply:
On the other other hand, I suppose making them make sure of that is something that would go against the views you expressed over in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46185#discussion_r457735856.

Reply:
So the `| null | undefined` suffix is actually correct. I agree this needs a neater approach if you're going to fix #20302 wholesale and I also agree that doing such a thing would be out of scope for this PR. But yes, don't make bugs worse in the meanwhile.

Reply:
It was pretty straightforward to add support for this, admittedly wishing that I had opted to tackle #20302 in this project now. Oh well, that can jus be a future project 😄